test/streams.js

```diff
@@ -3,121 +3,126 @@ var collect = require('stream-collector')
 var create = require('./helpers/create')
 var bufferFrom = require('buffer-from')
 
-tape('createReadStream to createWriteStream', function (t) {
-  var feed1 = create()
-  var feed2 = create()
-
-  feed1.append(['hello', 'world'], function () {
-    var r = feed1.createReadStream()
-    var w = feed2.createWriteStream()
-
-    r.pipe(w).on('finish', function () {
-      collect(feed2.createReadStream(), function (err, data) {
-        t.error(err, 'no error')
-        t.same(data, [bufferFrom('hello'), bufferFrom('world')])
-        t.end()
+function test (batch = 1) {
+  tape('createReadStream to createWriteStream', function (t) {
+    var feed1 = create()
+    var feed2 = create()
+
+    feed1.append(['hello', 'world'], function () {
+      var r = feed1.createReadStream({ batch })
+      var w = feed2.createWriteStream()
+
+      r.pipe(w).on('finish', function () {
+        collect(feed2.createReadStream({ batch }), function (err, data) {
+          t.error(err, 'no error')
+          t.same(data, [bufferFrom('hello'), bufferFrom('world')])
```

Comment:
Unrelated to this PR this made me realise we need this https://github.com/hypercore-protocol/hypercore/issues/265